### PR TITLE
Replace CLI mysql commands with PDO

### PR DESCRIPTION
**Pull request by balintsera**
Replaced command line mysql to pdo. Not sure how to continue, but now the installer works without mysql binary client (we use docker).

### CODE REVIEW

File: src/Joomlatools/Console/Command/Database/AbstractDatabase.php [hunk 1]
```diff
@@ -99,14 +100,42 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+        } catch (\PDOException $ex) {
+            throw new \RuntimeException(sprintf('Database query failed: error: "%s" "%s"',$ex->getMessage(), $query));
+        }
+    }
+
+    private function dbConnection() {
+        if (! $this->pdoDB) {
+            $password = empty($this->mysql->password) || $this->mysql->password == 'root' ? '' :  $this->mysql->password;
+            $connectionString = "mysql:host={$this->mysql->host}:{$this->mysql->port};dbname={$this->target_db};charset=utf8mb4";
+            var_dump($connectionString, $this->mysql->user, $password);
```
Comment (balintsera) [inline src/Joomlatools/Console/Command/Database/AbstractDatabase.php]: oops sorry, I will definitely remove this later (on the weekend)

File: src/Joomlatools/Console/Command/Database/Install.php [hunk 2]
```diff
@@ -92,25 +92,22 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
-            file_put_contents($tmp, $contents);
-
-            $password = empty($this->mysql->password) ? '' : sprintf("-p'%s'", $this->mysql->password);
-            $result = exec(sprintf("mysql --host=%s --port=%u --user='%s' %s %s < %s", $this->mysql->host, $this->mysql->port, $this->mysql->user, $password, $this->target_db, $tmp));
-
-            unlink($tmp);
-
+        foreach ($imports as $import) {
+            $content = file_get_contents($import);
+            if (strlen($content) < 2) {
```
Comment (stevenrombauts) [inline src/Joomlatools/Console/Command/Database/Install.php]: Why did you include this check here? 

We might want to consider adding a `file_exists()` check here though. But if the file length is less than 2 characters, it shouldn't be included in the first place :)

File: src/Joomlatools/Console/Command/Database/AbstractDatabase.php (hunk 1 above)
Comment (stevenrombauts) [inline src/Joomlatools/Console/Command/Database/AbstractDatabase.php]: No worries :)

File: src/Joomlatools/Console/Command/Database/Install.php (hunk 2 above)
Comment (balintsera) [inline src/Joomlatools/Console/Command/Database/Install.php]: it's just a remainder of an earlier solution, where I tried to read the file line-by-line with a generator, I will replace with file_exists()

**Comment by balintsera** [comment]
Done! 

Thank you for your warm welcome, let me know if I can polish this further somehow. We started to use the forked version, but will switch back after the merge.

**Comment by ercanozkaya** [comment]
Hi @balintsera, I know this is a bit of a late response but we decided to go with the mysql binary as we find it both faster and easier to work with. Thanks for the contribution! :)

**Pull request closed without merging**